src/server/service/kv.rs

```diff
@@ -914,48 +900,62 @@ impl<T: RaftStoreRouter + 'static, E: Engine> tikvpb_grpc::Tikv for Service<T, E
 }
 
 fn response_batch_commands_request<F>(
-    executor: &tokio_threadpool::Sender,
     id: u64,
     resp: F,
     tx: Sender<(u64, BatchCommandsResponse_Response)>,
     timer: HistogramTimer,
-    thread_load: Arc<ThreadLoad>,
 ) where
     F: Future<Item = BatchCommandsResponse_Response, Error = ()> + Send + 'static,
 {
-    let executor1 = executor.clone();
     let f = resp.and_then(move |resp| {
-        if tx.send((id, resp)).is_err() {
+        if tx.send_and_notify((id, resp)).is_err() {
             error!("KvService response batch commands fail");
             return Err(());
         }
         timer.observe_duration();
-        if let Some(notifier) = tx.get_notifier() {
-            if thread_load.in_heavy_load() {
```

Comment:
in_heavy_load will not used any more?

Reply:
It's still used to tell clients TiKV's load.